### ISSUE TITLE
schema.org URLs should resolve the HTTPS form

### ISSUE THREAD

**Issue by Aaranged** [schemaorg/schemaorg#716]
Currently any schema.org is equally retrievable as HTTP or HTTPS.  For example, both of these URLs return a 200 response header:
https://schema.org/Offer
http://schema.org/Offer

Long-standing best practices for canonicalization would see a page resolve under one, and only one, URL, with the non-canonical URL resolving to the canonical form.

This first and foremost because search engines index the HTTP and HTTPS versions, where both return a 200 (and where there are no certificate issues for HTTPS) as separate URLs.

While the issues that arise from permitting dual protocol URLs are, indeed, chiefly related to a sub-standard presence in the search engines (due to the generation of duplicate content, split link popularity, and similar issues), there's no demonstrable _benefit_ I know of for supporting dual protocol URLs.

It's worth noting, too, that because of their preference for HTTPS (http://bit.ly/1DtYcGA), Google has for some time being providing searchers with the HTTPS form of HTTP URLs, whereas Bing and Yandex (from a spot check) are directing searchers to the HTTP form - meaning that markup that originates on a copy/paste from the address bar following on a search is going to result in an arbitrary choice of protocols.

Accordingly, I propose that the HTTP form of schema.org URLs be 301 redirected to the HTTPS form.  An acceptable alternative (although one that would make Google a sad panda) would be 301 in the opposite direction.

The bottom line is that (at least IMO) schema.org shouldn't leave it to chance which version of a URL search engines serve to users, as there's no need for that ambiguity.

**Comment by BarbaraStarrSemanticFuse** [comment]
Could not agree more.

I second that proposal, as I am sure do many!  I have recently seen some significant errors in this regard.  (Which may have been due to other errors earlier today)

**Comment by scor** [comment]
:+1:  - note another related issue to redirect www.schema.org to schema.org: #4

**Comment by akuckartz** [comment]
The alternative - redirecting from https to http - likely violates data protection laws in some jurisdictions and should be considered a "worst practice".

**Comment by danbri** [comment]
This is waiting on some improvements to appengine to fix properly. Serving as https for a raw domain without www i.e. https://schema.org/Person or https://bib.schema.org/Person involves us in some special magic currently. In the meantime the schema.org partners have all agreed that they'll accept data that uses the https formulation of term names.

**Comment by RichardWallis** [comment]
http/https Still needs addressing as an overall issue pull (#751) however has introduced both consistency -- whichever version you arrive in a session it will stay in that style -- also as the software now knows if it is in http or https we have the basis for a final solution to this.

**Comment by RichardWallis** [comment]
Implemented
